rosbridge_server/src/rosbridge_server/websocket_handler.py

```diff
@@ -68,11 +66,37 @@ def wrapper(*args, **kwargs):
     return wrapper
 
 
-class RosbridgeWebSocket(WebSocketHandler):
+@implementer(interfaces.IPushProducer)
+class OutgoingValve:
+    """Allows the Autobahn transport to pause outgoing messages from rosbridge."""
```

Comment:
Does this detect back-pressure and pauses publishing internally?

Reply:
Yes, added some more description for this.